src/ibmca-engine-opensslconfig.in

```diff
@@ -0,0 +1,143 @@
+#!/usr/bin/perl
+
+#
+# Copyright 2022 International Business Machines Corp.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+#
+# Generate openssl.cnf from the system config file with added engine section
+# for ibmca engine.
+#
+# USE WITH CARE: No automation can replace human knowledge and understanding
+#                of the desired configuration!
+#
+
+use strict;
+use warnings;
+
+sub generate()
+{
+    my ($osslconfpath);
+    my ($tmp, $ih, $line, $oh, $defaultcnfsect, $indefaultsect, $enginesect);
+
+    $osslconfpath = `openssl version -d` || die "Please install openssl binary";
+    $osslconfpath =~ s/OPENSSLDIR: \"([^\"]*)\"$/$1/ || die "Failed to extract OpenSSL configuration directory";
+    chomp $osslconfpath;
+
+    open($ih, "<", "$osslconfpath/openssl.cnf") or die "Cannot open $osslconfpath/openssl.cnf";
+    open($oh, ">", "openssl.cnf.ibmca") or die "Cannot open openssl.cnf.ibmca";
+
+    $defaultcnfsect = undef;
+    $indefaultsect = 0;
+    $enginesect = undef;
+    while ($line = <$ih>) {
+	if ($line =~ /openssl_conf\s*=\s*(.*)/) {
```

Comment:
you use tabs here, no spaces. Is this intended?

Reply:
That is my editors default indentation for perl ;-)

Reply:
well its a wild mixture of tabs and blanks, thats why I mention it.  Either use tabs all over the place (although coding_style.md tells to use 4 blanks), or use 4 blanks per tab....

Reply:
untabbified...